### PR TITLE
more psv output

### PR DESCRIPTION
**Pull request by tranma**
needs some cleanup but outputting Pairs and Options seems to work

### CODE REVIEW

File: data/sea/50-psv.h [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #if !ICICLE_NO_PSV
 
+#define OUTPUT_BUF_SIZE 128
```
Comment (jystic) [inline data/sea/50-psv.h]: Maybe put this with the other constants a bit further down:
```c
/* psv driver */
static const size_t psv_max_row_count = 128;
static const size_t psv_buffer_size   = 16*1024;
```

File: src/Icicle/Sea/FromAvalanche/Psv.hs [hunk 2]
```diff
@@ -462,77 +463,159 @@ seaOfWriteProgramOutput state = do
...
         attrib      = seaOfEscaped name
+        free s      = indent 4 $ "free (" <> s <> ");"
+        calloc  n s = "char *" <> n <> " = calloc(1," <> s <> " );"
+        decli   n   = "int   " <> n <> " = 0;"
+    in  do (str, bufs) <- strOfOutput ps oname otype0 ts0 ixStart
+           case bufs of
+             ((b,_,_):_) -> pure
+                    $ vsep
+                    $    ["{" ]
+                      <> fmap (\(n,s,_) -> indent 4 $ calloc n s) (List.reverse bufs)
```
Comment (jystic) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: If you wanted you could allocate this out of the mempool which will avoid the function call overhead, also you don't need to worry about freeing it then, the gc will do it.

**Comment by jystic** [comment]
Looking good :+1:

File: src/Icicle/Sea/FromAvalanche/Psv.hs (hunk 2 above)
Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: oh cool! It's easy to free this though, maybe less work on the gc is better? dunno, see how outputting arrays goes

**Comment by tranma** [comment]
can output pairs of any nesting as a flat array

**Comment by amosr** [comment]
I like this. it's kind of silly, but you can say
```
feature asx ~> mean open, mean high, mean low
```
and operator precedence just nests it into a (Double,(Double,Double)) or maybe the other way around.
then if we just print that as ``[double, double, double]`` that's neat.

:+1:

**Comment by jystic** [comment]
> I like this. it's kind of silly, but you can say
> ```
> feature asx ~> mean open, mean high, mean low
> ```
> and operator precedence just nests it into a (Double,(Double,Double)) or maybe the other way around.
> then if we just print that as ``[double, double, double]`` that's neat.

yeah that seems like it would be quite useful!

**Comment by tranma** [comment]
this should be ready for final :eyes: 
then I'll try outputting arrays

**Comment by jystic** [comment]
:spaghetti:

**Pull request merged**